src/libos/src/vm/process_vm.rs

```diff
@@ -108,17 +108,73 @@ impl<'a, 'b> ProcessVMBuilder<'a, 'b> {
 
         // Now that we end up with the memory layout required by the process,
         // let's allocate the memory for the process
-        let process_range = {
-            // TODO: ensure alignment through USER_SPACE_VM_MANAGER, not by
-            // preserving extra space for alignment
-            USER_SPACE_VM_MANAGER.alloc(process_layout.align() + process_layout.size())?
-        };
+        // TODO: ensure alignment through USER_SPACE_VM_MANAGER, not by
+        // preserving extra space for alignment
+        let new_process_size = process_layout.align() + process_layout.size();
+
+        // Reuse old VM if possible
+        if let Some(old_vm) = reuse_process_vm {
+            let old_range = old_vm.process_range().range();
+            if old_range.size() >= new_process_size {
+                let mut vm_manager = old_vm.vm_manager();
+                // Clean VM manager
+                vm_manager.sanitize()?;
```

Comment:
If the reuse case "vfork", we should not sanitize the vm_manager.

Reply:
True. Removed.